bokehjs/test/protocol/message.ts

```diff
@@ -0,0 +1,179 @@
+import {expect} from "chai"
+
+import {Message} from "protocol/message"
+
+class MockSock {
+  readonly sent: unknown[] = []
+
+  send(data: unknown): void {
+    this.sent.push(data)
+  }
+}
+
+describe("protocol/message module", () => {
+
+  describe("Message", () => {
+
+    describe("assemble method", () => {
+
+      it("should create new Messages from JSON", () => {
+        const m = Message.assemble('{"msgid": "10", "msgtype": "FOO"}', '{"bar":2}', '{"baz":3}')
+        expect(m).to.be.instanceof(Message)
+        expect(m.complete()).to.be.true
+
+        expect(m.header).to.be.deep.equal({msgid: '10', msgtype: 'FOO'})
+        expect(m.metadata).to.be.deep.equal({bar:2})
+        expect(m.content).to.be.deep.equal({baz:3})
+        expect(m.buffers).to.be.deep.equal([])
+      })
+    })
+
+    describe("assemble_buffer method", () => {
+      const m = Message.create("FOO", {bar:2}, {baz:3})
+      m.header.num_buffers = 2
+
+      it("should append a new buffer", () => {
+        m.assemble_buffer(1, 2)
```

Comment:
@bryevdv, I suppose that arguments to `assemble_buffer()` are just whatever/throwaway values (they don't match types I figured out in `src/lib/protocol/message.ts`)?

Reply:
the values here in the test are just junk, yes